### PR TITLE
Add role annotation for workers

### PR DESCRIPTION
**Pull request by flaper87**
# Description

Worker BMH should have the role annotation set

## Type of change

Please select the appropriate options:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change is a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Versions:
- Hardware:

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not be merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

**Comment by achuzhoy** [comment]
/lgtm

Do we need one for master as well?

**Comment by flaper87** [comment]
> /lgtm
> 
> Do we need one for master as well?

no, masters are automatically assigned by assisted.

File: ztp/internal/data/cluster/objects/090-baremetalhost.yaml [hunk 1]
```diff
@@ -10,6 +10,9 @@ metadata:
   annotations:
     inspect.metal3.io: disabled
     bmac.agent-install.openshift.io/hostname: {{ .Hostname }}
+{{ if eq .Kind "Worker" }}
+    bmac.agent-install.openshift.io/role: worker
+{{ end }}
```
Comment (jhernand) [inline ztp/internal/data/cluster/objects/090-baremetalhost.yaml:15]: Would be nice if you can indent these lines:

```
    {{ if eq .Kind "Worker" }}
    bmac.agent-install.openshift.io/role: worker
    {{ end }}
```

It makes the template easier to read, in my opinion.

**Review by jhernand — COMMENTED** [review]
Just a minor comment about indentation, looks good.

File: ztp/internal/data/cluster/objects/090-baremetalhost.yaml (hunk 1 above)
Comment (flaper87) [reply ztp/internal/data/cluster/objects/090-baremetalhost.yaml:15]: Sorry, bad habit from helm templates. Let me do that

**Pull request merged**